examples/example-gateway/clients/baz/baz.go

```diff
@@ -66,11 +68,21 @@ func NewClient(config *zanzibar.StaticConfig, gateway *zanzibar.Gateway) *BazCli
 type SimpleServiceClient struct {
 	thriftService string
 	client        zt.TChanClient
+	// timeout is the default timeout if upstream does not have one
+	timeout time.Duration
 }
 
 // Call ...
 func (c *SimpleServiceClient) Call(ctx context.Context, reqHeaders map[string]string, args *baz.SimpleService_Call_Args) (map[string]string, *baz.BazResponse, error) {
 	var result baz.SimpleService_Call_Result
+
+	_, ok := ctx.Deadline()
```

Comment:
right, besides the default endpoint timeout, we probably should also have configurable timeouts for different endpoints and clients, thoughts?